index.js

```diff
@@ -0,0 +1,209 @@
+const typeScriptSettings = {
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    ecmaVersion: 2018,
+    ecmaFeatures: {
+      jsx: true,
+    },
+    sourceType: 'module',
+    project: './tsconfig.json',
+  },
+  extends: [
+    'plugin:@typescript-eslint/recommended',
+    'prettier/@typescript-eslint',
+    'plugin:prettier/recommended',
+  ],
+  rules: {
+    '@typescript-eslint/array-type': ['error', { default: 'generic', readonly: 'generic' }],
+    '@typescript-eslint/ban-ts-comment': 'error',
+    '@typescript-eslint/consistent-type-assertions': 'error',
+    '@typescript-eslint/explicit-member-accessibility': 'error',
+    '@typescript-eslint/indent': 'off',
+    '@typescript-eslint/naming-convention': [
+      'error',
+      {
+        selector: 'default',
+        format: ['camelCase'],
+        leadingUnderscore: 'forbid',
+        trailingUnderscore: 'forbid',
+      },
+      {
+        selector: 'typeLike',
+        format: ['PascalCase'],
+      },
+      {
+        selector: 'variable',
+        format: ['camelCase', 'UPPER_CASE'],
+      },
+      {
+        selector: 'enumMember',
+        format: ['PascalCase'],
+      },
+    ],
+    '@typescript-eslint/no-empty-function': 'error',
+    '@typescript-eslint/no-explicit-any': 'error',
+    '@typescript-eslint/no-inferrable-types': 'off',
+    '@typescript-eslint/no-unused-vars': [
+      'error',
+      { vars: 'all', args: 'after-used', ignoreRestSiblings: true },
+    ],
+    '@typescript-eslint/no-use-before-define': 'error',
```

Comment:
I agree with @ThaNarie when this rule is enabled for functions then the order is from least relevant to relevant. That doesn't make sense.

Reply:
We also need to decide what to do about the following:
```js
export const foo = () => {
  // not allowed, but works just fine as long as `foo` is not called before `bar` is defined
  bar();
}

export const bar = () => {
  // do something
}
```

```js
export function foo() {
  // allowed with the above setting
  bar();
}

export function foo() {
  // do something
}
```

I feel the `const foo = () => {}` currently is the preferred way of defining (and exporting) functions, both for normal utils as for react components. The downside is that those functions are not hoisted, and could potentially run into actual 'no-use-before-define` issues. However, when used in util files that just export functions, it should be totally safe.

If we want to keep using this rule, we should either switch to actual functions, or make sure our code is in the right order (which can be hard if function are conditionally recursive).

Reply:
@ThaNarie I think your second function in your second example should be called bar.

Just to add to the discussion. I'm a fan of having many files exporting a single function, and I generally use arrow functions for everything. I have worked with this rule on and I only remember having it complain about TS typing which can be out of order. But even then I think complying with the rule is fine. Having the program read in order makes removes surprises. If you want the important stuff on top, extract the non important stuff into other files.

Reply:
Sure, those are valid "workarounds" - just need to be careful that we don't force ourselves to do certain things different just because we want to use arrow functions and that rule enabled. A linter should help, not be (too much) in the way.

But yeah, let's try to export utils from its own file some more (what we already do with hooks anyway) - see how that goes.

Still a bit strange how we switched to using arrow functions, while normal functions work equally well, and have less downsides.